fix(slideout): Header updates [khcp-7919] (#1539)
Kong/kongponents: docs/components/slideout.md

Tells the component whether or not to render the open panel.

### closeButtonAlignment

Controls the close button alignment, can be `start` (default) or `end`.

### offsetTop

> Allows a host app to define the offset from the top of the page, it takes a `number` and that number uses `px` as the unit, defaults to `0px`.

Reviewer: ```suggestion
Allows a host app to define the offset from the top of the page, expects a `number` and that will be converted to `px`, defaults to `0px`.
```

suggestion

Tells the component whether or not to render the open panel.

### closeBtnAlignment

Controls the close button alignment, defaults to `start`.

> ### topOffsetStyle

Reviewer 2: this doesn't match the prop name in the component

        <p>Anim officia eiusmod duis est consequat nulla tempor ad non magna Lorem ullamco nostrud amet. Occaecat voluptate dolor enim eiusmod do qui nulla pariatur enim. Et elit elit consequat do do duis enim est ullamco id sunt sunt amet eiusmod. Do minim mollit irure ea sunt officia minim sint eiusmod enim amet. Quis exercitation in ullamco quis aliqua.</p>
        <p>Anim officia eiusmod duis est consequat nulla tempor ad non magna Lorem ullamco nostrud amet. Occaecat voluptate dolor enim eiusmod do qui nulla pariatur enim. Et elit elit consequat do do duis enim est ullamco id sunt sunt amet eiusmod. Do minim mollit irure ea sunt officia minim sint eiusmod enim amet. Quis exercitation in ullamco quis aliqua.</p>
        <p>Anim officia eiusmod duis est consequat nulla tempor ad non magna Lorem ullamco nostrud amet. Occaecat voluptate dolor enim eiusmod do qui nulla pariatur enim. Et elit elit consequat do do duis enim est ullamco id sunt sunt amet eiusmod. Do minim mollit irure ea sunt officia minim sint eiusmod enim amet. Quis exercitation in ullamco quis aliqua.</p>
      </div>
      <KButton appearance="primary">Buttons</KButton>

>       <KCard class="mt-2">

Author: I have already updated this class for the token work, will be updated here once rebased.

        <template v-slot:body>
          Something in a KCard
        </template>
      </KCard>
      <KAlert
        appearance="info"
        alert-message="Or even an alert!"
        class="mt-2" />
    </KSlideout>
  </div>
</KToggle>
```
### hasOnlyTitle

> Determines whether the header will display only `Title` or has other attributes like `Badge`, `Icon` etc, defaults to `false`.

Author: removed this prop.

        <p>Anim officia eiusmod duis est consequat nulla tempor ad non magna Lorem ullamco nostrud amet. Occaecat voluptate dolor enim eiusmod do qui nulla pariatur enim. Et elit elit consequat do do duis enim est ullamco id sunt sunt amet eiusmod. Do minim mollit irure ea sunt officia minim sint eiusmod enim amet. Quis exercitation in ullamco quis aliqua.</p>
      </div>
      <KButton appearance="primary">Buttons</KButton>
      <KCard class="mt-2">
        <template v-slot:body>
          Something in a KCard
        </template>
      </KCard>
      <KAlert
        appearance="info"
        alert-message="Or even an alert!"
        class="mt-2" />
    </KSlideout>
  </div>
</KToggle>
```

### badgeValue

This prop takes a string that will be displayed as the value for the `Badge`.

### badgeAppearance

Determines the appearance values for the Badge, defaults to `warning`.

### badgeShape

Determines the shape of the Badge, defaults to `rectangular`.

### iconType

Determines the icon type that is placed next to Title, defaults to `copy`.

### iconColor

Determines the icon color that is placed next to Title, defaults to `blue`.

> ### iconSize

Reviewer: issue: We are actually trying to move away from this type of approach with too many props customizing optional parts of the design. It would be much cleaner to simply offer a single slot following the title maybe call it `title-after` and let the consuming app be responsible for rendering the badge/icon if it's needed.

</KSlideout>
```

> - `badgeContent` - used to customize `badge` inside the header

Reviewer: suggestion to use three slots: `title-before`, `title`, and `title-after` as discussed above

- `titleContent` - used to customize `title` inside the header
- `iconContent` - used to customize `icon` inside the header

## Events

- `close` - Emitted when the close button is clicked, anything outside the panel is clicked, or the `esc` key is pressed.

> - `kclipboard-icon-clicked` - Emitted when the `icon` next to `title` is clicked.

Reviewer: suggestion: if the icon content is slottable we should use a more generic event (if we even need it) - if we allow the consuming app to completely control the content via slots, we no longer need to handle/emit the click event.

Author: Yes, removed this.
